assets/js/base/context/add-to-cart-form-context.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import {
+	createContext,
+	useContext,
+	useState,
+	useCallback,
+} from '@wordpress/element';
+import {
+	useStoreAddToCart,
+	useTriggerFragmentRefresh,
+} from '@woocommerce/base-hooks';
+
+/**
+ * @typedef {import('@woocommerce/type-defs/contexts').AddToCartFormContext} AddToCartFormContext
+ */
+
+const AddToCartFormContext = createContext( {
```

Comment:
I've listed events and emitters I think we'll need but I will most likely need some help (in the follow up) making these work - it should be similar to the checkout extensibility work.